packages/libs/react-ui/src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,21 +1,47 @@
-import React, { forwardRef } from 'react';
-import { arrowVariants, container } from './Tooltip.css';
-
-export interface ITooltipProps {
-  placement?: 'top' | 'bottom' | 'left' | 'right';
+import { Box } from '@components/Layout';
+import type { FC } from 'react';
+import React, { cloneElement, useRef } from 'react';
+import { useTooltip, useTooltipTrigger } from 'react-aria';
+import type { TooltipTriggerProps } from 'react-stately';
+import { useTooltipTriggerState } from 'react-stately';
+import { tooltipPositionVariants } from './Tooltip.css';
+export interface ITooltipProps extends Omit<TooltipTriggerProps, 'trigger'> {
   children: React.ReactNode;
+  content: React.ReactNode;
+  position?: keyof typeof tooltipPositionVariants;
 }
 
-export const Tooltip: React.ForwardRefExoticComponent<
-  Omit<ITooltipProps, 'ref'> & React.RefAttributes<HTMLDivElement>
-  // eslint-disable-next-line react/display-name
-> = forwardRef<HTMLDivElement, ITooltipProps>(
-  ({ children, placement = 'right' }, ref) => {
-    return (
-      <div className={container} ref={ref} data-placement={placement}>
-        <div className={arrowVariants[placement]} />
-        {children}
-      </div>
-    );
-  },
-);
+export const Tooltip: FC<ITooltipProps> = ({
+  children,
+  content,
+  position = 'right',
+  ...props
+}) => {
+  const config = {
+    delay: 500,
+    closeDelay: 300,
+    ...props,
+  };
+  const state = useTooltipTriggerState(config);
+  const ref = useRef(null);
+
+  const { triggerProps, tooltipProps: baseTooltipProps } = useTooltipTrigger(
+    config,
+    state,
+    ref,
+  );
+
+  const { tooltipProps } = useTooltip(baseTooltipProps, state);
+
+  return (
+    <Box position="relative">
+      {cloneElement(children as React.ReactElement, { ...triggerProps, ref })}
```

Comment:
didnt we want to avoid using this? or is this the only way to pass the props?

Reply:
We want to use this sparingly and in a lot of previous cases, using context was better, but in this case it is the simplest option since we don't want the user to have to access a context